changeset.go

```diff
@@ -188,6 +219,10 @@ func (cs *Changeset) Status(ctx context.Context, changesetNamespace, changesetNa
 
 	return retry(ctx, retryAttempts, retryPeriod, func() error {
 		for _, op := range tr.Spec.Items {
+			if id != "" && op.ID != id {
```

Comment:
What does this check for?

Reply:
If the intent here is to differentiate b/w checking the status of a particular operation vs the entire changeset, then I'd do something like this:

```go
const AllOperations = "*"

func (cs *Changeset) Status(ctx context.Context, chNamespace, chName, opID string, ...) ... {
  ...
  return retry(ctx, retryAttempts, retryPeriod, func() error {
    for _, op := range tr.Spec.Items {
      if opID != AllOperations && opID != op.ID {
        continue
      }
    }
  }
}

// To check overall changeset status.
cs.Status(ctx, namespace, name, AllOperations, ...)
```

Or better yet, I'd see if it's possible to have a separate method for checking particular operation status.

Reply:
Let's add the Status API for the specific operation.